indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSourceTest.java

```diff
@@ -135,6 +137,20 @@ public void testReadTimeout() throws IOException
     Assert.assertTrue(supplier.isClosed());
   }
 
+  @Test
+  public void testRecordSupplierInputSourceThrowsSamplerExceptionWhenExceptionDuringSeek()
+  {
+    final RecordSupplier<?, ?, ?> supplier = Mockito.mock(RecordSupplier.class);
+    Mockito.when(supplier.getPartitionIds("test-stream"))
+           .thenThrow(new StreamException(new Exception("Something bad happened")));
+
+    //noinspection ResultOfObjectAllocationIgnored
+    Assert.assertThrows(
```

Comment:
Maybe also verify (part of) the exception message.

Reply:
Done.